src/UI/Art/SpriteArtist/PlayerSetupColorFactionArtist.java

```diff
@@ -0,0 +1,116 @@
+package UI.Art.SpriteArtist;
+
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+
+import Engine.IController;
+import UI.PlayerSetupColorFactionController;
+import UI.UIUtils;
+import Units.UnitModel;
+
+public class PlayerSetupColorFactionArtist
+{
+  // Keep track of all the unit images we need to juggle.
+  private static BufferedImage[][] unitArray;
+  private static int unitSizePx; // Units are square
+  private static int unitBuffer;
+  private static int numFactions;
+  private static int numColors;
+  private static int gridWidth;
+  private static int gridHeight;
+
+  public static void draw(Graphics g, IController controller)
+  {
+    PlayerSetupColorFactionController control = (PlayerSetupColorFactionController)controller;
+    if( null == control )
+    {
+      System.out.println("WARNING! PlayerSetupColorFactionArtist was given the wrong controller!");
+    }
+
+    // Make sure we have all the things we need to draw.
+    initialize();
+
+    int color = control.getSelectedColor();
+    int faction = control.getSelectedFaction();
+
+    // Get the draw space
+    int drawScale = SpriteOptions.getDrawScale();
+    Dimension dimensions = SpriteOptions.getScreenDimensions();
+    int myWidth = dimensions.width / drawScale;
+    int myHeight = dimensions.height / drawScale;
+
+    // Start by assuming we can center the unit grid.
+    int startX = myWidth / 2 - gridWidth / 2;
+    int startY = myHeight / 2 - gridHeight / 2;
+
+    // Figure out if we need to realign things.
+    int maxX = myWidth - myWidth/3;
+    int maxY = myHeight - myHeight/3;
+    int minX = myWidth/3;
+    int minY = myHeight/3;
```

Comment:
It's... kinda weird to have a third of your screen potentially empty. I'm thinking shave that down to 1/8th or something, since it feels like a lot of wasted real estate.
Also, the buffers aren't evenly wide since you don't draw the units centered.

Reply:
Yeah, I initially had it top-left justified, but it felt too out-of-the way putting it that far off. Making it semi-centered felt a lot more natural, but this is definitely tweakable.

Reply:
I still think I want like 1/7th or 1/8th or something in that ballpark. The margins feel a little too large still IMO.